dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -473,7 +480,13 @@ def read_partition(
                 pf._dtypes = lambda *args: pf.dtypes  # ugly patch, could be fixed
                 pf.fmd.row_groups = None
             rg_piece = pf.row_groups[piece]
-            pf.fmd.key_value_metadata = None
+            if null_index_name:
+                if "__index_level_0__" in pf.columns:
```

Comment:
You may wish to add comments here, to explain this.
It may be nice to upstream this to fastparquet, not urgent.